tests/test_host_check.py

```diff
@@ -486,14 +505,67 @@ def test_plat_specified_check_failing(self, capsys):
 ==================================================================
 !! Host NOT set up correctly for xrd-control-plane !!
 ==================================================================
+"""
+        assert output == cli_output
+        assert exit_code == 1
+
+    def test_plat_specified_check_erroring(self, capsys):
+        """Test a platform check erroring when host-check is run with arguments."""
+        exit_code, output = self.run_host_check(
+            capsys,
+            ["-p", "xrd-control-plane"],
+            erroring_checks=["Kernel version"],
+            failing_checks=["Kernel version"],
+        )
+        cli_output = f"""\
+==============================
+Platform checks - xrd-control-plane
+==============================
+Checks: {CONTROL_PLANE_CHECKS_STR}
+
+==================================================================
+!! One or more checks could not be performed, see errors above !!
+==================================================================
+"""
+        assert output == cli_output
+        assert exit_code == 1
+
+    def test_base_check_erroring(self, capsys):
+        """Test a base check erroring when host-check is run without arguments."""
+        exit_code, output = self.run_host_check(
+            capsys,
+            [],
+            erroring_checks=["Kernel version"],
+            failing_checks=["Kernel version"],
+        )
+        cli_output = f"""\
+==============================
+Platform checks
+==============================
+
+base checks
+-----------------------
+Checks: {BASE_CHECKS_STR}
+
+xrd-control-plane checks
+-----------------------
+Checks: RAM
+
+xrd-vrouter checks
+-----------------------
+Checks: CPU extensions, RAM, Hugepages, Interface kernel driver, IOMMU, Shared memory pages max size, Real-time Group Scheduling
+
+==================================================================
+!! One or more checks could not be performed, see errors above !!
+==================================================================
 """
         assert output == cli_output
         assert exit_code == 1
 
```

Comment:
Missing testcases:
- error from an "extra check"
- error for a platform-specific check with no args (mirror of `test_one_plat_supported()`)